### ISSUE TITLE
Support Teckton Pipelines

### ISSUE THREAD

**Issue by fkorotkov** [cirruslabs/cirrus-ci-docs#316]
# Description

[Teckton Pipelines](https://github.com/tektoncd/pipeline) are useful primitives that can help to implement #85 and use different Docker containers for each script command.

[Teckton's Task syntax](https://github.com/tektoncd/pipeline/blob/master/docs/tutorial.md#task) in not particularly appealing so let's use it syntax under the hood but still use Cirrus CI's syntax for defining scripts, caches and so on.

Proposed syntax for such tasks can look like this:

```yaml
task:
  container:
    image: node:latest
  install_script: npm ci
  build_script: npm build
  use:
    container:
      dockerfile: .ci/Dockerfile
  validate_links_script: liche --document-root=site --recursive site/ 
```

Ideally we would have syntax to start one task from the part where another task has finished with syntax like this:

```yaml
build_task:
  container:
    image: node:latest
  install_script: npm ci
  build_script: npm build

validate_task:
  continue_from: build
  container:
    dockerfile: .ci/Dockerfile
  validate_links_script: liche --document-root=site --recursive site/ 
```

But in this situation is not clear how to do re-runs. What if `validate` task fails and needs a rerun. We'll need to re-run `build` task as well. 🤔

**Comment by fkorotkov** [comment]
Another syntax option is to introduce new type of tasks along the regular `task`s and `docker_builder` tasks and call it `pipe`. A `pipe` will have the same fields as a `task` or a `docker_builder` but will require to specify `steps` list field in order to specify instructions to execute and containers to execute these instructions in:

```yaml
pipe:
  name: "Build and Validate Links"
  resources:
    cpu: 2
    memory: 2G
  auto_cancelation: true
  steps:
    - image: squidfunk/mkdocs-material:latest
      install_script: pip install --upgrade pymdown-extensions
      build_script: mkdocs build
    - dockerfile: .ci/Dockerfile
      validate_script: liche --document-root=site --recursive site/
```

**Comment by fkorotkov** [comment]
Fixed in #338 but without using Teckton Pipelines 🤷‍♂
